tests/manage/pv_services/test_rwo_pvc_fencing_unfencing.py

```diff
@@ -0,0 +1,632 @@
+                        label_key="cluster.ocs.openshift.io/openshift-storage",
+                        label_value="''"
+                    )
+                else:
+                    # Todo: use add_node function when implemented
+                    pytest.skip("Insufficient nodes, Add node support needed")
+
+        else:
+            if len(non_ocs_nodes_name) < num_of_nodes:
+                pytest.skip("Insufficient nodes, Add node support needed")
+
+        logger.info(f"Worker Nodes:\n{worker_nodes_name}")
+        logger.info(f"OCS Nodes:\n{ocs_nodes_name}")
+        logger.info(f"Non-OCS Nodes:\n{non_ocs_nodes_name}")
+        return worker_nodes_name, ocs_nodes_name, non_ocs_nodes_name
+
+    def select_nodes_for_app_pods(
+        self, scenario, ocs_nodes, non_ocs_nodes, num_of_nodes,
+        ignore_check=False
+    ):
+        """
+        Select nodes for the running app pods
+        Colocated scenario: Select 1 OCS node where osd and/or mon is running,
+            select other nodes where mon/osd are not running
+        Dedicated scenario: Select non-OCS nodes
+
+        Args:
+            scenario (str): Scenario of app pods running on OCS or dedicated nodes
+                (eg., 'colocated', 'dedicated')
+            ocs_nodes (list): list of OCS nodes name
+            non_ocs_nodes (list): list of non-OCS nodes name
+            num_of_nodes (int): number of nodes to be selected
+            ignore_check (bool): True for selecting OCS nodes without checking
+                for mon and osd pods, False otherwise
+
+        Returns:
+            app_pod_nodes_name (list): list of names of selected nodes
+        """
+        app_pod_nodes_name = []
+        # Todo: 3AZ Colocated scenario - select nodes from AZ having more than one worker node
+        if scenario == "colocated":
+            if ignore_check:
+                app_pod_nodes_name += ocs_nodes[0:num_of_nodes]
+            else:
+                mon_pods = pod.get_mon_pods()
+                mon_pod_nodes_name = [pod.get_pod_node(pod_obj).name for pod_obj in mon_pods]
+                osd_pods = pod.get_osd_pods()
+                osd_pod_nodes_name = [pod.get_pod_node(pod_obj).name for pod_obj in osd_pods]
+                common_node = list(set(mon_pod_nodes_name) & set(osd_pod_nodes_name))
+
+                if common_node:
+                    app_pod_nodes_name.append(common_node[0])
+                else:
+                    app_pod_nodes_name.append(osd_pod_nodes_name[0])
+
+                if num_of_nodes > 1:
+                    available_nodes = list(set(ocs_nodes) - set(mon_pod_nodes_name + osd_pod_nodes_name))
+                    app_pod_nodes_name += available_nodes[0:num_of_nodes - 1]
+
+        else:
+            app_pod_nodes_name += non_ocs_nodes[0:num_of_nodes]
+
+        return app_pod_nodes_name
+
+    def setup_pods_run_ios(self, pod_list, fio_filename='io_file'):
```

Comment:
any chance we have a function with such implementation in other tests?
If not, thinking it would be good to take this out of the test file to some commonplace

Reply:
Similar implementations are present in other tests but they slightly differ from one another.
For eg. [test_change_reclaim_policy_of_pv.py#L83](https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/pv_services/test_change_reclaim_policy_of_pv.py#L83), ceph_daemon_kill and resource_deletion related tests in pv_services